### PR TITLE
Add solution blog post for 2025 day 9

### PR DESCRIPTION
**Pull request by stewSquared**

### CODE REVIEW

File: docs/2025/puzzles/day09.md [hunk 1]
```diff
@@ -2,11 +2,189 @@ import Solver from "../../../../../website/src/components/Solver.js"
...
 
 https://adventofcode.com/2025/day/9
 
+## Solution Summary
+
+We use rectangle representations and search over all possible rectangles for the maximum, filtering for part 2 by checking interesections with boundary lines.
+
+### Part 1
+
+For Part 1, it suffices to calculate the area for all possible rectangles, but modelling this nicely will help with Part 2, so we'll take advantage of some of the tools Scala gives us. We'll use case classes for `Point` and `Area`. An Area is a set representing a rectangular grid of points -- the 3D analog of a [`Range`](https://www.scala-lang.org/api/current/scala/collection/immutable/Range.html). Like a Range, it can function as a virtual collection.
```
Comment (merlinorg) [inline docs/2025/puzzles/day09.md]:
```suggestion
For Part 1, it suffices to calculate the area for all possible rectangles, but modelling this nicely will help with Part 2, so we'll take advantage of some of the tools Scala gives us. We'll use case classes for `Point` and `Area`. An Area is a set representing a rectangular grid of points -- the 2D analog of a [`Range`](https://www.scala-lang.org/api/current/scala/collection/immutable/Range.html). Like a Range, it can function as a virtual collection.
```

File: docs/2025/puzzles/day09.md [hunk 2]
```diff
@@ -2,11 +2,189 @@ import Solver from "../../../../../website/src/components/Solver.js"
...
+
+## Alternatives and Optimizations
+
+A point can be determined to be inside or outside the boundary in linear time by counting the number of lines between the edge of the grid and the point. An odd number of boundary crossings means the tile is green. This can then be used to test every point inside a candidate rectangle and would detect the edge cases missed above.
+
+The number of checks can be reduced by using edge compression on the coordinates, so that entire rectangles of tiles that have no overlaps with any red tile coordinate can have their color determined at the same time. Alternatively, one could optimize by using a disjoint set data structure. The `Area` type above can be used like a set with O(1) set membership, and the full set of green tiles could be represented by a collection of these.
+
+Any alternative approaches take significantly more work, however. [@merlinorg](https://github.com/merlinorg/) has provided [an example](https://github.com/merlinorg/advent-of-code/blob/44a80dd81d54cea13255e4013ad28cf18fbfbb8e/src/main/scala/year2025/day09alt.scala) that fully handles all edge cases.
```
Comment (merlinorg) [inline docs/2025/puzzles/day09.md]:
```suggestion
Any alternative approaches take significantly more work, however. [@merlinorg](https://github.com/merlinorg/) has provided [an example](https://github.com/merlinorg/advent-of-code/blob/44a80dd81d54cea13255e4013ad28cf18fbfbb8e/src/main/scala/year2025/day09alt.scala) that handles more, but still not all, edge cases.
```

**Review by merlinorg — APPROVED** [review]
Nice.

File: docs/2025/puzzles/day09.md [hunk 3]
```diff
@@ -2,11 +2,192 @@ import Solver from "../../../../../website/src/components/Solver.js"
...
+
+### Edge Cases that don't need to be handled
+
+The shape of the input data is largely a circle, with a single rectangular incurion through its center. Because of this, there are two classes of edge cases that don't need to be handled.
+
+First, when we check that a rectangle doesn't intersect any boundaries, that does not actually tell us the rectangle is fully green. It tells us that the rectangle is either fully inside or outside of the circle. Because of the shape of the input, the largest non-intersected rectangle will always be fully contained.
+
+Second, boundaries can intersect our rectangle without diminishing the number of green tiles, as long as it's directly adjacent to another boundary.
+
+## Alternatives and Optimizations
```
[the remark this answers is not in the document]
Comment (stewSquared) [reply docs/2025/puzzles/day09.md]: Alright, so range-compressed flood fill from the outside rather than the inside. Yeah, I can modify the section a bit to mention that. It's similar to what I mean by using a disjoint set data structure.

**Comment by SethTisue** [comment]
@stewSquared looks like we shouldn't merge quite yet, one piece of pending work?

**Review by Philippus — APPROVED** [review]
